src/webots/nodes/WbGeometry.cpp

```diff
@@ -524,6 +524,12 @@ int WbGeometry::triangleCount() const {
     return 0;
 }
 
+bool WbGeometry::exportNodeHeader(WbVrmlWriter &writer) const {
+  if (writer.isUrdf())
+    return true;
+  return WbNode::exportNodeHeader(writer);
```

Comment:
No reason, both have the same effect at this point and I didn't know what practice is better. Now I know, thanks :)

Reply:
Ok, in that case calling the closest parent is better as it might be dangerous to skip nodes in the hierarchy. With the current code, if in the future we override `exportNodeHeader` in `WbBaseNode`, then it will not be called from `WbGeometry` and this is not very easy to detect, so to avoid future issues we prefer to call the closest parent.